jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -77,6 +103,8 @@
     String _uri;
     String _decodedPath;
 
+    boolean _ambiguousSegment;
```

Comment:
This class implements `equals()` but not `hashCode()`! Must be implemented using the same fields as `equals()`.